editoast/src/schema/speed_section.rs

```diff
@@ -15,21 +15,44 @@ use derivative::Derivative;
 use editoast_derive::InfraModel;
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
+use validator::{Validate, ValidationError};
 
-#[derive(Debug, Derivative, Clone, Deserialize, Serialize, PartialEq, InfraModel)]
+#[derive(Debug, Derivative, Clone, Deserialize, Serialize, PartialEq, InfraModel, Validate)]
 #[serde(deny_unknown_fields)]
 #[infra_model(table = "crate::tables::infra_object_speed_section")]
 #[derivative(Default)]
 pub struct SpeedSection {
     pub id: Identifier,
     #[derivative(Default(value = "Some(80.)"))]
+    #[validate(custom = "validate_speed_limit")]
     pub speed_limit: Option<f64>,
+    #[validate(custom = "validate_speed_limit_by_tag")]
     pub speed_limit_by_tag: HashMap<NonBlankString, f64>,
     pub track_ranges: Vec<ApplicableDirectionsTrackRange>,
     #[serde(default)]
     pub extensions: SpeedSectionExtensions,
 }
 
+pub const SPEED_LIMIT_ERROR: &str = "speed_limit must be greater than 0";
```

Comment:
It looks like this is only useful for testing purposes. We could only check that the validation fails and inline this (keeping it private).